chia/farmer/farmer.py

```diff
@@ -835,3 +835,59 @@ async def _periodically_clear_cache_and_refresh_task(self) -> None:
                 log.error(f"_periodically_clear_cache_and_refresh_task failed: {traceback.format_exc()}")
 
             await asyncio.sleep(1)
+
+    def notify_farmer_reward_taken_by_harvester_as_fee(
+        self, sp: farmer_protocol.NewSignagePoint, proof_of_space: harvester_protocol.NewProofOfSpace
+    ) -> None:
+        """
+        Apply a fee quality convention (see CHIP-22: https://github.com/Chia-Network/chips/pull/88)
+        given the proof and signage point. This will be tested against the fee threshold reported
+        by the harvester (if any), and logged.
+        """
+        assert proof_of_space.farmer_reward_address_override is not None
+
+        challenge_str = str(sp.challenge_hash)
+
+        ph_prefix = self.config["network_overrides"]["config"][self.config["selected_network"]]["address_prefix"]
+        farmer_reward_puzzle_hash = encode_puzzle_hash(proof_of_space.farmer_reward_address_override, ph_prefix)
+
+        self.log.info(
+            f"Farmer reward for challenge '{challenge_str}' "
+            + f"taken by harvester for reward address '{farmer_reward_puzzle_hash}'"
+        )
+
+        fee_quality = calculate_harvester_fee_quality(proof_of_space.proof.proof, sp.challenge_hash)
+        fee_quality_rate = int(float(fee_quality) / float(0xFFFFFFFF) * 100.0)
```

Comment:
this metric seems a bit odd to me. it's not really a percent. it's just the "quality", which could be mapped onto a scale of 0-100

Reply:
That is what the line is doing, mapping it to 0-100

Reply:
(Or did I misunderstand something?)

Reply:
it doesn't really matter. it just seems odd to present this value as a fraction, in percent. But the more I think about it, the less odd it seems